catalogd/internal/storage/localdir.go

```diff
@@ -20,95 +26,266 @@ import (
 // done so that clients accessing the content stored in RootDir/catalogName have
 // atomic view of the content for a catalog.
 type LocalDirV1 struct {
-	RootDir string
-	RootURL *url.URL
+	RootDir            string
+	RootURL            *url.URL
+	EnableQueryHandler bool
+
+	m  sync.RWMutex
+	sf singleflight.Group
 }
 
-const (
-	v1ApiPath = "api/v1"
-	v1ApiData = "all"
+var (
+	_                Instance = &LocalDirV1{}
```

Comment:
Nit: it is _slightly_ better to do the compile-time interface check this way to avoid an allocation:
```suggestion
	_                Instance = (*LocalDirV1)(nil)
```